feat(curriculum) add type coercion debug lab (#62930)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/blocks/lab-debug-type-coercion-errors/68f890b73bbfc9efa7379e7b.md

---
id: 68f890b73bbfc9efa7379e7b
title: Debug Type Coercion Errors in a Buggy App
challengeType: 26
dashedName: debug-type-coercion-errors-in-a-buggy-app
---

# --description--

You’ve just joined a local web development shop, and your first assignment is to clean up some buggy code left behind by the previous developers.

> They were attempting simple arithmetic operations, but something went wrong, the results don’t make sense. Your task is to review, debug, and correct the code so it performs as expected and is easier to read.

Reviewer: ```suggestion
They were attempting simple arithmetic operations, but something went wrong, the results don't make sense. Your task is to review, debug, and correct the code so it performs as expected and is easier to read.
```

# --hints--

You should have a variable called `firstResult`.

```js
assert.match(code, /const\s+firstResult\s*=/);
```

`firstResult` should have the numeric value `15`.

```js
assert.strictEqual(firstResult, 15);
```

You should have a variable called `secondResult`.

```js
assert.match(code, /const\s+secondResult\s*=/);
```

`secondResult` should have the numeric value `3`.

```js
assert.strictEqual(secondResult, 3);
```

You should have a variable called `thirdResult`.

```js
assert.match(code, /const\s+thirdResult\s*=/);
```

`thirdResult` should have the numeric value `6`.

```js
assert.strictEqual(thirdResult, 6);
```

`thirdResult` should not include a boolean value (`true` or `false`).

```js
assert.notMatch(code, /true|false/);
```

You should have a variable called `fourthResult`.

```js
assert.match(code, /const\s+fourthResult\s*=/);
```

`fourthResult` should have the numeric value `8`.

```js
assert.strictEqual(fourthResult, 8);
```

`fourthResult` should not include a boolean value (`true` or `false`).

```js

> assert.notMatch(code, /true|false/);

Reviewer: Same here

```

You should have a variable called `fifthResult`.

```js
assert.match(code, /const\s+fifthResult\s*=/);
```

`fifthResult` should have the numeric value `20`.

```js
assert.strictEqual(fifthResult, 20);
```

`fifthResult` should not include a string multiplication (e.g. "10" * 2).

```js
assert.notMatch(code, /".*"\s*\*/);
```

You should have a variable called `sixthResult`.

```js
assert.match(code, /const\s+sixthResult\s*=/);
```

`sixthResult` should have the numeric value `22`.

```js
assert.strictEqual(sixthResult, 22);
```

`sixthResult` should not include `null` in its expression.

```js

> assert.notMatch(code, /null/);

Reviewer: and here

`thirdResult` should not include a boolean value (`true` or `false`).

```js
{
  const match = code.match(/const\s+thirdResult\s*=\s*([^;]+)/);
  assert.isNotNull(match);
  const expression = match[1];
  assert.notMatch(expression, /\btrue\b|\bfalse\b/);
}
```

You should have a variable called `fourthResult`.

```js
assert.match(code, /const\s+fourthResult\s*=/);
```

`fourthResult` should have the numeric value `8`.

```js
assert.strictEqual(fourthResult, 8);
```

`fourthResult` should not include a boolean value (`true` or `false`).

```js
{
  const match = code.match(/const\s+fourthResult\s*=\s*([^;]+)/);
  assert.isNotNull(match);
  const expression = match[1];
  assert.notMatch(expression, /\btrue\b|\bfalse\b/);
}
```

You should have a variable called `fifthResult`.

```js
assert.match(code, /const\s+fifthResult\s*=/);
```

`fifthResult` should have the numeric value `20`.

```js
assert.strictEqual(fifthResult, 20);
```

`fifthResult` should not include a string multiplication (e.g. "10" * 2).

```js
{
  const match = code.match(/const\s+fifthResult\s*=\s*([^;]+)/);
  assert.isNotNull(match);
  const expression = match[1];
  assert.notMatch(expression, /["'`].*["'`]\s*\*/);

> }

Reviewer: ```suggestion
const match = code.match(/const\s+fifthResult\s*=\s*([^;]+)/);
assert.isNotNull(match);
const expression = match[1];
assert.notMatch(expression, /["'`].*["'`]\s*\*/);
```

```

You should have a variable called `sixthResult`.

```js
assert.match(code, /const\s+sixthResult\s*=/);
```

`sixthResult` should have the numeric value `22`.

```js
assert.strictEqual(sixthResult, 22);
```

`sixthResult` should not include `null` in its expression.

```js
{
  const match = code.match(/const\s+sixthResult\s*=\s*([^;]+)/);
  assert.isNotNull(match);
  const expression = match[1];
  assert.notMatch(expression, /\bnull\b/);

> }

Reviewer: ```suggestion
const match = code.match(/const\s+sixthResult\s*=\s*([^;]+)/);
assert.isNotNull(match);
const expression = match[1];
assert.notMatch(expression, /\bnull\b/);
```
